ion/binarywriter.go

```diff
@@ -591,13 +626,14 @@ func (w *binaryWriter) end(api string, t ctx) error {
 
 // Resolve resolves a symbol to its ID.
 func (w *binaryWriter) resolve(api, sym string) (uint64, error) {
-	if strings.HasPrefix(sym, "$") {
-		id, err := strconv.ParseUint(sym[1:], 10, 64)
-		if err == nil {
-			return id, nil
-		}
+	if id, ok := symbolIdentifier(sym); ok {
```

Comment:
I want to understand the symbol ID interpretation behavior that we're implementing here. I'm having difficulty imagining a scenario where I'm using a binary writer and I pass it a string in the `$id` format (`$12`, `$47`, etc).